network/irc/irc.go

```diff
@@ -596,6 +614,29 @@ func (bot *ircBot) act(theLine *line.Line) {
 		return
 	}
 
+	isAskingForSasl := theLine.User == "" && strings.HasSuffix(theLine.Raw, " CAP * ACK :sasl")
+	if isAskingForSasl {
+		bot.sendSaslStart()
+		return
+	}
+
+	isAskingForSaslPass := theLine.User == "" && theLine.Raw == "AUTHENTICATE +"
+	if isAskingForSaslPass {
+		bot.sendSaslPass()
+		return
+	}
+
+	isSaslConfirm := theLine.User == "" && theLine.Content == "SASL authentication successful"
```

Comment:
Checking for `theLine.Command == "903"` would be safer here. The message content may vary.